packages/editor-ui/src/components/NodeSettings.vue

```diff
@@ -43,705 +39,369 @@ import { get, set, unset } from 'lodash-es';
 
 import NodeExecuteButton from './NodeExecuteButton.vue';
 import { isCommunityPackageName } from '@/utils/nodeTypesUtils';
-import { useUIStore } from '@/stores/ui.store';
 import { useWorkflowsStore } from '@/stores/workflows.store';
 import { useNDVStore } from '@/stores/ndv.store';
 import { useNodeTypesStore } from '@/stores/nodeTypes.store';
 import { useHistoryStore } from '@/stores/history.store';
 import { RenameNodeCommand } from '@/models/history';
-import useWorkflowsEEStore from '@/stores/workflows.ee.store';
 import { useCredentialsStore } from '@/stores/credentials.store';
 import type { EventBus } from 'n8n-design-system';
 import { useExternalHooks } from '@/composables/useExternalHooks';
 import { useNodeHelpers } from '@/composables/useNodeHelpers';
-import { importCurlEventBus, ndvEventBus } from '@/event-bus';
 import { useToast } from '@/composables/useToast';
+import { useI18n } from '@/composables/useI18n';
+import { useTelemetry } from '@/composables/useTelemetry';
+import { importCurlEventBus, ndvEventBus } from '@/event-bus';
 
-export default defineComponent({
-	name: 'NodeSettings',
-	components: {
-		NodeTitle,
-		NodeCredentials,
-		ParameterInputList,
-		NodeSettingsTabs,
-		NodeWebhooks,
-		NDVSubConnections,
-		NodeExecuteButton,
-	},
-	setup() {
-		const nodeHelpers = useNodeHelpers();
-		const externalHooks = useExternalHooks();
-		const { showMessage } = useToast();
-
-		return {
-			externalHooks,
-			nodeHelpers,
-			showMessage,
-		};
-	},
-	computed: {
-		...mapStores(
-			useHistoryStore,
-			useNodeTypesStore,
-			useNDVStore,
-			useUIStore,
-			useCredentialsStore,
-			useWorkflowsStore,
-			useWorkflowsEEStore,
-		),
-		isReadOnly(): boolean {
-			return this.readOnly || this.hasForeignCredential;
-		},
-		isExecutable(): boolean {
-			if (this.nodeType && this.node) {
-				const workflow = this.workflowsStore.getCurrentWorkflow();
-				const workflowNode = workflow.getNode(this.node.name);
-				const inputs = NodeHelpers.getNodeInputs(workflow, workflowNode!, this.nodeType);
-				const inputNames = NodeHelpers.getConnectionTypes(inputs);
-
-				if (!inputNames.includes(NodeConnectionType.Main) && !this.isTriggerNode) {
-					return false;
-				}
-			}
+const props = defineProps<{
```

Comment:
We should add defaults for the ones that have them using `withDefaults`. E.g. `foreignCredentials` default was `[]`, `blockUI` had `false`, etc.